---
title: "Re-analysis of De Los Angeles et al. (2024) bioRxiv TPM data"
author: "Sam Buckberry"
date: "`r Sys.Date()`"
output: 
  html_document:
    code_folding: hide
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

### Data acquistion from De Los Angeles et al.
Analysis code and data from De Los Angeles et al. was made available to us through GitHub and Synapse (data repository)

https://github.com/labsyspharm/tnt-rebuttal   
https://www.synapse.org/#!Synapse:syn53061839/files/  

For the re-analysis below, we downloaded the following files:  
`biosample_meta.csv`    
`salmon_tpm.csv.gz`  

and saved them to the directory `dla-data`.  


### Re-analysis of gene expression estimates generated by De Los Angeles et al. 

First, let's plot timecourse of reprogramming for Sendai expression for samples shown in Buckberry _et al._ (2023) _Nature_. Here, we are only including the samples analysed in our paper (Donor 32F) as these are the ones we used in generating the TNT hypothesis, and only used the t2iLGoY naive media throughout this paper. This also allows us to simplify the data visualisation to make our point, and only plot together samples from the same experimental setup and design.  

```{r, message=FALSE, error=FALSE, comment=FALSE, warning=FALSE, cache=TRUE}
## Source the packages and functions we use in this analysis
source("project-functions.R")

## Load the RNA-seq quant data generated by De Los Angeles et al 
sample_meta <- read.csv("dla-data/biosample_meta.csv")
salmon_quants <- read.csv("dla-data/salmon_tpm.csv.gz")

##Subset data for Sendai genes
sev_ind <- str_starts(salmon_quants$transcript_id, fixed("Sendai"))
salmon_quants <- salmon_quants[sev_ind, ]

## Join the metadata with the quant data
ind <- match(salmon_quants$sample_accession, sample_meta$sample_accession)

sev_dat <- cbind(salmon_quants, sample_meta[ind, ])
sev_dat <- sev_dat[sev_dat$transcript_id != "Sendai_F_gene", -8]
```


```{r, message=FALSE, error=FALSE, comment=FALSE, warning=FALSE, cache=TRUE, fig.height=2}
liu_dat <- sev_dat[sev_dat$study_name %in% 
                         c("Liu Nature 2020", "Liu Nat Met 2017"), ]
liu_dat <- liu_dat[liu_dat$time_point %in% 
                       c("P8", "P20 + 3", "D3", "D7", "D13", "D21", "P3", "P10"), ]

liu_dat$time_point <- factor(liu_dat$time_point,
                               levels=c("D3", "D7", "D13", "D21", "P3", "P10"))

liu_dat <- liu_dat[liu_dat$treatment %in% c("Fibroblast", "Primed", "t2iLGoY"), ]

liu_dat <- liu_dat[!((liu_dat$age == "55") & !is.na(liu_dat$age)), ]


liu_dat <- liu_dat[!is.na(liu_dat$time_point), ]

## Subset for media analysed in Buckberry et al
reprog_pal <- c(Primed="#009E73", t2iLGoY="#0072B2", Fibroblast="#CC79A7",
                TNT="#EEBC4C", NtP="#55B3EA", "#5B7876", "#AFB4B7")

plot_sendai_timecourse <- function(transcript_id, legend_pos = "none"){
    
    gg <- ggplot(data = liu_dat[liu_dat$transcript_id == transcript_id, ],
                 mapping = aes(x = time_point, y = tpm,
                               group=treatment, colour=treatment, fill=treatment)) +
        geom_point(size=2, alpha=0.7) + 
        stat_summary(geom = "line") +
        ggtitle(transcript_id) +
        sams_pub_theme(legend_pos = legend_pos) +
        scale_color_manual(values = reprog_pal[c("Fibroblast", "t2iLGoY", "Primed")])
    gg    
}

gg_timecourse <- lapply(X = unique(liu_dat$transcript_id), FUN = plot_sendai_timecourse)

cowplot::plot_grid(plotlist = gg_timecourse, nrow = 1)
```

We can see in the figure above that there is a significant drop in Sendai virus expression from day 3 of reprogramming. Notably, the largest difference between Sendai gene expression for primed and naive reprogramming occurs at day 13, with ~2-fold less Sendai virus in naive media. This is a noteworthy observation, as TNT reprogramming transitions the cells from naive media to primed media at the day 13 time point, and challenges the speculation that Sendai virus persistence impacts TNT reprogramming.  


Now let's plot Sendai virus expression for MEL1 isogenic reprogramming experiments. Note that this is the only experiment set from Buckberry _et al._ (2023) _Nature_ where we performed RNA-seq on TNT-iPS cells.  

```{r, fig.height=2, fig.width=8, message=FALSE, error=FALSE, comment=FALSE, cache=TRUE, warning=FALSE}
mel1_dat <- sev_dat[sev_dat$background %in% c("MEL1"), ]
mel1_dat <- mel1_dat[!is.na(mel1_dat$transcript_id), ]

reprog_pal2 <- c(TNT="#eebc4cff", Primed="#009e73ff",
                 NtP="#55b3eaff", "Control ESC" ="#a3a3a3ff",
                 Na誰ve="#0072b2ff", Fibroblast="#cc79a7ff")

plot_sendai_endpoint <- function(transcript_id, legend_pos = "none"){
    
    gg <- ggplot(data = mel1_dat[mel1_dat$transcript_id == transcript_id, ],
                 mapping = aes(x = treatment, y = tpm,
                               group=treatment, colour=treatment, fill=treatment)) +
        geom_point(size=2, alpha=0.7) + 
        ggtitle(transcript_id) +
        sams_pub_theme(legend_pos = legend_pos) +
        scale_colour_manual(values = reprog_pal2)
    gg    
    
}

gg_mel1 <- lapply(X = unique(mel1_dat$transcript_id), FUN = plot_sendai_endpoint)

cowplot::plot_grid(plotlist = gg_mel1, nrow = 1)

```


We can clearly see in the plot above that Sendai virus expression in Naive-iPS cells is far higher than any other sample.    

Now let's plot these two experiment sets these together on the same scale, and then the MEL1 isogenic reprogramming experiments on their own scale so we have a comprehensive picture of the range of Sendai expression across these two experiment sets.   


```{r, fig.width=7, fig.height=7, message=FALSE, error=FALSE, comment=FALSE, cache=TRUE, warning=FALSE}

## Set-up some clear labels
plot_dat <- rbind(liu_dat, mel1_dat)
plot_dat$group <- ifelse(plot_dat$background == "MEL1", yes = "MEL1", no = "32F")
plot_dat$group[is.na(plot_dat$group)] <- "32F"

plot_dat$id <- NA
plot_dat$id[plot_dat$group == "MEL1"] <- plot_dat$treatment[plot_dat$group == "MEL1"]
plot_dat$id[plot_dat$group == "32F"] <- as.character(plot_dat$time_point[plot_dat$group == "32F"])

plot_dat$treatment[plot_dat$treatment == "Na誰ve"] <- "Naive (t2iLGoY)"
plot_dat$treatment[plot_dat$treatment == "t2iLGoY"] <- "Naive (t2iLGoY)"
plot_dat$treatment[plot_dat$treatment == "t2iLGoY"] <- "Naive (t2iLGoY)"

plot_dat$id[plot_dat$id == "Na誰ve"] <- "Naive (t2iLGoY)"

plot_dat$id <- factor(plot_dat$id, levels = c("D3", "D7", "D13", "D21", "P3", "P10",
                                              "Control ESC", "Fibroblast", 
                                              "Naive (t2iLGoY)", "TNT", "NtP", "Primed"))

reprog_pal <- c(Primed="#009E73", "Naive (t2iLGoY)"="#0072B2", Fibroblast="#CC79A7",
                TNT="#EEBC4C", NtP="#55B3EA", "Control ESC" ="#a3a3a3ff")

#reprog_pal2 <- c(TNT="#eebc4cff", Primed="#009e73ff",
#                 NtP="#55b3eaff", "Control ESC" ="#a3a3a3ff",
#                 Na誰ve="#0072b2ff", Fibroblast="#cc79a7ff")


plot_dat$transcript_id <- str_replace(plot_dat$transcript_id, pattern = "Sendai_", "") %>%
    str_replace(pattern = "_", " ")

sams_pub_theme2 <- function(legend_pos = "none", x.text.angle = 45, hjust = 1,
                           y.text.angle = 0, line_point = 0.5) {
    line_mm <- line_point / 2.835
    custom_theme <- theme_bw() +
        theme(plot.background = element_blank(),
              panel.grid.minor = element_blank(),
              panel.grid.major = element_line(size = line_mm),
              panel.border = element_rect(color = "black", size = line_mm, fill = NA), # Add border around each facet
              axis.text.x = element_text(angle = x.text.angle, hjust = hjust, size = 8, colour = 'black'),
              axis.text.y = element_text(size = 8, colour = 'black', angle = y.text.angle),
              strip.text.y = element_text(size = 8),
              text = element_text(size = 8),
              #strip.background = element_blank(),
              legend.position = legend_pos,
              axis.line.x = element_line(color = 'black', size = line_mm),
              axis.line.y = element_line(color = 'black', size = line_mm),
              axis.ticks = element_line(color = 'black', size = line_mm))
    return(custom_theme)
}


gg <- ggplot(data = plot_dat, mapping = aes(x = id, y = tpm,
                           group=treatment, colour=treatment, fill=treatment)) +
    geom_point(size=2, alpha=0.7) + 
    stat_summary(geom = "line") +
    facet_grid(transcript_id~group, space = "free_x", drop = TRUE, scales = "free") +
    sams_pub_theme2(legend_pos = "none") +
    xlab(label = "") + ylab("TPM") +
    scale_colour_manual(values = reprog_pal)

gg_mel1 <- ggplot(data = plot_dat[plot_dat$group == "MEL1", ],
                  mapping = aes(x = id, y = tpm,
                           group=treatment, colour=treatment, fill=treatment)) +
    geom_point(size=2, alpha=0.7) + 
    stat_summary(geom = "line") +
    facet_grid(transcript_id~group, space = "free_x", drop = TRUE, scales = "free") +
    sams_pub_theme2(legend_pos = "right") +
    xlab(label = "") + ylab("TPM") +
    scale_colour_manual(values = reprog_pal)

cowplot::plot_grid(gg, gg_mel1, align = "h", rel_widths = c(0.55, 0.5))
```

```{r save-plot, include=FALSE}
pdf("de-los-angeles-tpm-replot.pdf", width = 7, height = 7)
cowplot::plot_grid(gg, gg_mel1, align = "h", rel_widths = c(0.55, 0.5))
dev.off()

```

Although not visible in these plots, there is some very low-level detection of Sendai virus RNA-seq reads in samples that are not Naive-iPS cells for the MEL1 isogenic reprogramming system, and De Los Angeles et al. base their case on these findings. To transparently show the Sendai expression levels for these samples, we will create a table with all the details.   


```{r, message=FALSE, error=FALSE, comment=FALSE, cache=TRUE, warning=FALSE}

mel1_dat <- plot_dat[plot_dat$group == "MEL1", c("transcript_id", "time_point",
                                                  "sample_accession",
                                                  "count", "tpm", "title",
                                                  "treatment")]

# Data transformation
df <- mel1_dat %>%
  mutate(combined = paste(treatment, sample_accession, sep = "_")) 

# Pivot the data to wide format
wide_df <- df %>%
  tidyr::pivot_wider(names_from = transcript_id, values_from = tpm, id_cols = c(treatment, sample_accession))

wide_df <- wide_df[order(wide_df$treatment), ]

## Calc geometric mean
# geometric_mean <- function(x) {
#   exp(mean(log(x[x > 0])))  # Log-transform, average, and exponentiate back
# }
# 
# wide_df$SeV_geometric_mean <- lapply(X = 1:nrow(wide_df),
#                            function(x){
#                                geometric_mean(
#                                wide_df[x, 3:ncol(wide_df)])
#                                }
#                            ) %>% unlist()
# 
# wide_df$SeV_geometric_mean[!is.finite(wide_df$SeV_geometric_mean)] <- 0

wide_df <- wide_df %>%
  mutate(across(where(is.numeric), round, digits=2))



colnames(wide_df) <- str_remove(colnames(wide_df), "Sendai_") %>% str_remove(pattern = "sample_")

colnames(wide_df)[1] <- "Group"

wide_df$Group <- str_remove(wide_df$Group, "Control ")

colnames(wide_df) <- str_replace_all(string = colnames(wide_df), pattern = "_", replacement = " ")

knitr::kable(wide_df, booktabs = TRUE, latex_options = "scale_down") %>%
  kableExtra::kable_styling() %>%
  kableExtra::add_header_above(header = c(" " = 2, "Sendai gene expression (TPM)" = ncol(wide_df) - 2)) %>%
    kableExtra::add_footnote(notation = "symbol",
                             label = "TPM estimates generated by De Los Angeles et al. Original data from Buckberry et al. (2023) Nature")

    
```

In the table above, we can see that there is some-low level read counts in both ESC, one NtP, and one Primed line and none in TNT-iPS cells. 

Of note, there is considerable difference in the TPM values for each Sendai gene, thus it was probably inappropriate for De Los Angeles et al. to be simply averaging expression across all Sendai genes for their Fig 1b plot. 

However, the detection of Sendai virus in ESC samples warrants further investigation, which we will address in subsequent analyses.  


```{r}
sessionInfo()
```

